libs/hdf-converters/src/utils/attestations.ts

```diff
@@ -49,14 +48,38 @@ export function advanceDate(
     case 'daily':
       date.add(1, 'day');
       break;
-    default:
-      const msTime = ms(frequency);
-      if (!msTime) {
-        throw new Error('Unknown date format: ' + frequency);
-      } else {
-        date.add(msTime, 'milliseconds');
+    default: {
+      // a number followed by d/w/m/y, with or without spaces in between
+      const re = /(\d{1,10}(?:.\d{0,10})?)(\s{0,10})([a-z])/;
+      const match = re.exec(frequency);
+
+      if (!match) {
+        throw new Error(
+          'Unknown date format: ' +
+            frequency +
+            '. Please use a number followed by d/w/m/y to indicate days, weeks, months, or years, e.g. 1d/2w/3m/1y/custom.'
+        );
+      }
+
+      const number = match[1];
+      const unit = match[3];
```

Comment:
How does the 1 and 3 index here work? I want to make sure we are not making any assumptions about number of days or spaces or something, which may be taken care of by the regex, but I'm not sure. For example, '120 d' or '15       w'

Reply:
So the regular expression allows for 0-10 number of spaces between the number and the unit, and the space (or lack of spaces) would be match[2], and match [1] is the number and match [3] is the unit which also have character limits because the pipeline kept saying it was a security concern as somebody could enter a ton of zeros and DDoS the system